src/Graphics/Widgets/BoundingBoxWidget.h

```diff
@@ -29,21 +29,107 @@ DEALINGS IN THE SOFTWARE.
+        BOX,
+        CORNER_SCALE,
+        FACE_ROTATE,
+        FACE_SCALE,
+        X_PLUS,
+        Y_PLUS,
+        Z_PLUS,
+        X_MINUS,
+        Y_MINUS,
+        Z_MINUS };
 
-      class SCISHARE BoundingBoxWidget : public WidgetBase
+      class SCISHARE BoundingBoxWidget : public CompositeWidget
       {
       public:
-        BoundingBoxWidget(const Core::GeometryIDGenerator& idGenerator, double scale,
-          const BoxPosition& pos, const Core::Geometry::Point& origin, const Core::Geometry::BBox& bbox);
+        BoundingBoxWidget(const Core::GeometryIDGenerator& idGenerator, const std::string& name,
+                          double scale, const Core::Geometry::Point& center,
+                          const std::vector<Core::Geometry::Vector>& eigvecs,
+                          const std::vector<double>& eigvals, int widget_num);
+        BoundingBoxWidget(const Core::GeometryIDGenerator& idGenerator, const std::string& name,
+                          double scale, const BoxPosition& pos, const Core::Geometry::Point& center,
+                          int widget_num, const Core::Geometry::BBox& bbox);
+        BoundingBoxWidget(const Core::GeometryIDGenerator& idGenerator, const std::string& name,
+                          double scale, const Core::Geometry::Transform& trans,
+                          const Core::Geometry::Point& center, int widget_num);
+
+      private:
+        int RESOLUTION_ = 10;
+        const int DIMENSIONS_ = 3;
+        const int CORNERS_ = 8;
+        const int FACES_ = 6;
+        const float BOX_SCALE_ {1.0};
+        const float ROT_SPHERE_SCALE_ {2.0};
+        const float RESIZE_SPHERE_SCALE_ {1.5};
+        const float DISK_WIDTH_ {3.0};
+        const float DISK_RADIUS_ {1.0};
+        const std::string DEFL_POINT_COL_ {Core::Datatypes::ColorRGB(0.54, 0.6, 1.0).toString()};
+        const std::string DEFL_COL_ {Core::Datatypes::ColorRGB(0.5, 0.5, 0.5).toString()};
+        const std::string RESIZE_COL_ {Core::Datatypes::ColorRGB(0.54, 1.0, 0.60).toString()};
+        const std::string DISK_COL_ = RESIZE_COL_;
+        std::vector<WidgetHandle> translateWidgets_;
+        std::vector<WidgetHandle> rotateWidgets_;
+        std::vector<WidgetHandle> scaleWidgets_;
+        std::vector<std::vector<std::vector<WidgetHandle>>> scaleAxisWidgets_;
+        std::vector<std::string> allIds_;
+        std::vector<std::string> translateIds_;
+        std::vector<std::vector<std::vector<std::string>>> translateIdsByFace_;
+        std::vector<std::vector<std::string> > translateIdsBySide_;
+        std::vector<std::vector<std::vector<std::string>>> rotateIdsByFace_;
+        std::vector<std::vector<std::vector<std::string>>> scaleIdsByFace_;
+        std::vector<std::string> rotateIds_;
+        std::vector<std::string> scaleIds_;
+        std::vector<std::vector<std::vector<std::string>>> scaleAxisIds_;
+        std::string widgetName(size_t i, size_t id, size_t iter);
```

Comment:
As a blanket review comment, this is just way, way too many new private member variables. Come up with some classes that can break down this giant list, and give them names. No vector<vector<vector either.

Reply:
https://refactoring.guru/smells/large-class